include/jemalloc/internal/pai.h

```diff
@@ -8,6 +8,9 @@ struct pai_s {
 	/* Returns NULL on failure. */
 	edata_t *(*alloc)(tsdn_t *tsdn, pai_t *self, size_t size,
 	    size_t alignment, bool zero);
+	/* Allocate extent with page guards (page aligned only). */
```

Comment:
Agreed. I think the only reason I went this way, was to avoid changing the existing call sites, as I wanted to minimize the diff scope. Not valid anymore. Will update.